ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/MultiLineLoopRuleTest.kt

```diff
@@ -0,0 +1,229 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.test.KtLintAssertThat.Companion.assertThatRule
+import com.pinterest.ktlint.test.LintViolation
+import org.junit.jupiter.api.Test
+
+class MultiLineLoopRuleTest {
+    private val multiLineLoopRuleAssertThat = assertThatRule { MultiLineLoopRule() }
+
+    @Test
+    fun `Given loop statements with curly braces on single line`() {
+        val code =
+            """
+            fun foo() {
+                for (i in 1..10) { return 0 }
+                while (true) { return 0 }
+                do { return 0 } while (true)
```

Comment:
The test are all fine. But they would look a bit better when you do not use the `return 0` statement inside the body of the loop because in real code it would exit at the first iteration. Just use `bar()` to indicate that something is done which is not relevant for the test.